File: app/src/main/java/com/zxcx/zhizhe/ui/my/selectAttention/SelectAttentionModel.java
package com.zxcx.zhizhe.ui.my.selectAttention;

import android.support.annotation.NonNull;

import com.zxcx.zhizhe.mvpBase.BaseModel;
import com.zxcx.zhizhe.mvpBase.BaseRxJava;
import com.zxcx.zhizhe.retrofit.AppClient;
import com.zxcx.zhizhe.retrofit.BaseArrayBean;
import com.zxcx.zhizhe.retrofit.BaseBean;
import com.zxcx.zhizhe.retrofit.BaseSubscriber;
import com.zxcx.zhizhe.retrofit.NullPostSubscriber;

import java.util.List;

public class SelectAttentionModel extends BaseModel<SelectAttentionContract.Presenter> {

    public SelectAttentionModel(@NonNull SelectAttentionContract.Presenter present) {
        this.mPresenter = present;
    }

    public void getAttentionList(){
        mDisposable = AppClient.getAPIService().getAttentionList()
                .compose(BaseRxJava.<BaseArrayBean<SelectAttentionBean>>io_main())
                .compose(BaseRxJava.<SelectAttentionBean>handleArrayResult())
                .subscribeWith(new BaseSubscriber<List<SelectAttentionBean>>(mPresenter) {
                    @Override
                    public void onNext(List<SelectAttentionBean> list) {
                        mPresenter.getDataSuccess(list);
                    }
                });
        addSubscription(mDisposable);
    }

    public void changeAttentionList(List<Integer> idList){
        mDisposable = AppClient.getAPIService().changeAttentionList(idList)
                .compose(BaseRxJava.handlePostResult())
                .compose(BaseRxJava.<BaseBean>io_main_loading(mPresenter))
                .subscribeWith(new NullPostSubscriber<BaseBean>(mPresenter) {
                    @Override
                    public void onNext(BaseBean bean) {
                        mPresenter.postSuccess();
                    }
                });
        addSubscription(mDisposable);
    }
}

File: app/src/main/java/com/zxcx/zhizhe/ui/my/selectAttention/SelectAttentionModel.kt
package com.zxcx.zhizhe.ui.my.selectAttention

import com.zxcx.zhizhe.mvpBase.BaseModel
import com.zxcx.zhizhe.mvpBase.BaseRxJava
import com.zxcx.zhizhe.retrofit.AppClient
import com.zxcx.zhizhe.retrofit.BaseBean
import com.zxcx.zhizhe.retrofit.BaseSubscriber
import com.zxcx.zhizhe.retrofit.NullPostSubscriber
import com.zxcx.zhizhe.ui.classify.ClassifyBean

class SelectAttentionModel(present: SelectAttentionContract.Presenter) : BaseModel<SelectAttentionContract.Presenter>() {

	init {
		this.mPresenter = present
	}

	fun getClassify() {
		mDisposable = AppClient.getAPIService().classify
				.compose(BaseRxJava.io_main())
				.compose(BaseRxJava.handleArrayResult())
				.subscribeWith(object : BaseSubscriber<MutableList<ClassifyBean>>(mPresenter) {
					override fun onNext(list: MutableList<ClassifyBean>) {
						mPresenter?.getDataSuccess(list)
					}
				})
		addSubscription(mDisposable)
	}

	fun changeAttentionList(idList: List<Int>) {
		mDisposable = AppClient.getAPIService().changeAttentionList(idList)
				.compose(BaseRxJava.handlePostResult())
				.compose(BaseRxJava.io_main_loading(mPresenter))
				.subscribeWith(object : NullPostSubscriber<BaseBean<*>>(mPresenter) {
					override fun onNext(bean: BaseBean<*>) {
						mPresenter?.postSuccess()
					}
				})
		addSubscription(mDisposable)
	}
}


